generators/common/files.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2013-2018 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The default is to use a file path string. It implies use of the template method.
+ * For any other config an object { file:.., method:.., template:.. } can be used
+ */
+const commonFiles = {
+    global: [
+        {
+            templates: [
+                '.prettierrc', // this needs to be the first file for prettier transform to work
```

Comment:
You should take a look to https://github.com/jhipster/generator-jhipster/commit/d7ea3ded31472cf06f4ebc4e33bb5a28e11cf0f3#diff-7855c55ad2fef8b13dba21314afe58d4 if you want your prettier configuration to be applied on the generated files

Reply:
This is working using the current configuration and the prettier configuration is applied as expected. Using the default phase is to ensure the prettier config files are written prior to all the others ?

Reply:
> 
>
> Using the default phase is to ensure the prettier config files are written prior to all the others ?

Exactly. Because the `common` subgenerator is composed after the `server` and `client` subgenerators.
That's why I'm surprised that it works with your changes. Are you sure that the indentation is correct (2 spaces, not 4 spaces)? You might want to add a unit test to verify it:
https://github.com/jhipster/generator-jhipster/commit/d7ea3ded31472cf06f4ebc4e33bb5a28e11cf0f3#diff-c01b3a6fb20de17b2b4e257759937692R140

Reply:
I added Husky and corresponding test. It should be ok now

Reply:
I'm not really convinced by the way it works, but let's say it's ok.